bot/extensions/fun/weather.py

```diff
@@ -0,0 +1,104 @@
+from geopy.geocoders import Nominatim
+from timezonefinder import TimezoneFinder
+from pytz import country_timezones, timezone
+from datetime import datetime
+from discord.ext.commands import Cog, command
+from requests import get
+from discord import Embed
+from string import capwords
+
+
+class WeatherCog(Cog, name="Weather", description="get current weather information from a city"):
+    def __init__(self, bot):
+        self.bot = bot
+    
+    @command(name='weather', help='Show weather information in your city', usage="{city}")
+    async def weather(self, ctx, *city_input):
+        city = capwords(" ".join(city_input))
+
+        # initialize Nominatim API
+        geolocator = Nominatim(user_agent="geoapiExercises")
+        
+        # getting Latitude and Longitude
+        location = geolocator.geocode(city)
+
+        # pass the Latitude and Longitude
+        # into a timezone_at
+        # and it return timezone
+        timezone_finder = TimezoneFinder()
+
+        result = timezone_finder.timezone_at(
+            lng=location.longitude, 
+            lat=location.latitude)
+        timezone_city = datetime.now(timezone(result))
+        timezone_city = timezone_city.strftime('%m/%d/%Y %H:%M')
+
+        api_key = "441df3a5cadc2498e093c0367cae6817"
```

Comment:
You could put the base url as a static attribute, it would make the code a bit cleaner and more reusable in the future. (Optional for now)